google/cloud/internal/openssl_util.cc

```diff
@@ -31,6 +40,118 @@ GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
 namespace internal {
 
 namespace {
+#ifdef _WIN32
+std::string CaptureWin32Errors() {
+  auto last_error = GetLastError();
+  LPSTR message_buffer_raw = nullptr;
+  auto size = FormatMessageA(
+      FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM |
+          FORMAT_MESSAGE_IGNORE_INSERTS,
+      nullptr, last_error, MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT),
+      (LPSTR)&message_buffer_raw, 0, nullptr);
+  std::unique_ptr<char, decltype(&LocalFree)> message_buffer(message_buffer_raw,
+                                                             &LocalFree);
+  return std::string(message_buffer.get(), size) + " (error code " +
+         std::to_string(last_error) + ")";
+}
+
+StatusOr<std::vector<BYTE>> DecodePem(std::string const& pem_contents) {
+  DWORD buffer_size = 0;
+  if (!CryptStringToBinaryA(
+          pem_contents.c_str(), static_cast<DWORD>(pem_contents.size()),
+          CRYPT_STRING_BASE64HEADER, nullptr, &buffer_size, nullptr, nullptr)) {
+    return Status(StatusCode::kInvalidArgument,
+                  "Invalid ServiceAccountCredentials - could not parse PEM to "
+                  "get private key: " +
+                      CaptureWin32Errors());
```

Comment:
Here and elsewhere, please use 

https://github.com/googleapis/google-cloud-cpp/blob/c4921e481c985751b854fcb0efe7b00e1d21ef00/google/cloud/internal/make_status.h#L30

```suggestion
    return InvalidArgumentError(
                  absl::StrCat("Invalid ServiceAccountCredentials - could not parse PEM to "
                  "get private key: ",
                      CaptureWin32Errors(), GCP_ERROR_INFO());
```

Reply:
Done.